compiler/runtime/OMRCodeCacheManager.cpp

```diff
@@ -237,6 +237,7 @@ OMR::CodeCacheManager::destroy()
       self()->freeMemory(codeCache);
       codeCache = nextCache;
       }
+   self()->freeCodeCacheSegment(_codeCacheRepositorySegment);
```

Comment:
Does the codeCacheRepository always exist? In OpenJ9 sometimes (when the virtual address space is small) we do not allocate a repository. Instead, each code cache will have their own segment. Maybe adding a check is safer.

Reply:
In the JitBuilder and the Test Compiler, the code cache repository always exists. However, the option to not use a repository is available at the OMR level. So, you are correct, it would probably be safer to add a check. I will open a new PR with a fix.